src/timeline/InputBar.cpp

```diff
@@ -630,6 +630,14 @@ InputBar::command(QString command, QString args)
                 notice(args, false);
         } else if (command == "rainbownotice") {
                 notice(args, true);
+        } else if (command == "goto") {
+                // Goto has three different modes:
+                // 1 - Matrix URI handler, as if you clicked the URI
+                // 2 - Going directly to a given event ID
+                // 3 - Going directly to a given message index
+                if (!ChatPage::instance()->handleMatrixUri(args)) {
```

Comment:
I think checking if it starts with a $ or does not have a / is cleaner here?

Reply:
This was just the shortest code to write to get the wanted effect. Any kind of internally handleable URL works, and if it cannot handle it as a URL it goes to the "go to event" handler.  That handler checks for a beginning `$` character, and tries to parse as integer otherwise.

I've really enjoyed being able to do `/goto 0` to quickly scroll down using this ^_^